### PR TITLE
fix(home): Transfer button enables when at least one wallet can send (#199, #302)

### LINKED ISSUE #199: Transfer button on Home is greyed out (disabled style) even when 2+ wallets exist — and tapping it still opens the sheet
## What I saw

On Home with **2+ wallets present** (the wallet carousel shows page dots), the Transfer button — which should be enabled in that state — appears in the **disabled visual style**: the pink circular background is faded compared to Receive / Send either side of it. **Yet tapping it still opens the Transfer sheet.**

So two bugs in one:

1. **The visual `actionButtonDisabled` style is applied when it shouldn't be.** With ≥2 wallets and at least one that can send, Transfer is a valid action — the button should render in the fully-saturated enabled style.
2. **`disabled` prop and the style are out of sync.** Whatever's making the style disabled isn't making the button truly non-interactive; `onPress` still fires and the sheet opens.

Either outcome alone would be a small bug; having them disagree means the gating logic has two independent conditions that can diverge.

## Screenshot

Home on Pixel 8 (Android 16), 2 carousel dots visible, balance 57,318 sats, Transfer faded between Receive and Send — but tap opens the sheet.

## Relevant code

`src/screens/HomeScreen.tsx`:

```ts
const hasSendableWallet = wallets.some(
  (w) =>
    (w.walletType === 'nwc' && w.isConnected) ||
    (w.walletType === 'onchain' && w.onchainImportMethod === 'mnemonic'),
);
const canTransfer = hasSendableWallet && wallets.length >= 2;

<TouchableOpacity
  style={[styles.actionButton, !canTransfer && styles.actionButtonDisabled]}
  onPress={() => setTransferOpen(true)}
  disabled={!canTransfer}
  ...
>
```

Given the observed behaviour (disabled style + tap still fires), the likely shape is:
- `canTransfer === true` → `disabled={false}` → tap opens sheet ✓
- AND something ELSE is ALSO applying `actionButtonDisabled` to the style array, OR the conditional destructuring is evaluating stale wallet state.

It may also be that `hasSendableWallet` is too strict (only NWC-connected or onchain-mnemonic counts). This is the same class of bug we fixed for Send/Receive in #195's bundle: post-PR-D, NWC wallets land with `isConnected: false` and flip to true in background, so any gate hanging off transient `isConnected` will intermittently misfire.

## Expected

When 2+ wallets exist and at least one can send:
- Button renders in the fully-saturated enabled style
- Tap opens the Transfer sheet

When gating is truly false (e.g. only one wallet, or all watch-only):
- Button is greyed AND unresponsive

## Repro

- Device: Pixel 8, Android 16
- Build: v1.0.0-rc1 + #185 merged (local EAS production APK from commit `76a43e7`)
- Two wallets configured (NWC + at least one other)
- Steps: Home tab → observe Transfer button state + tap

## Ask

1. Log `canTransfer`, `hasSendableWallet`, `wallets.length`, and each wallet's `(walletType, isConnected, onchainImportMethod)` once on render — nail down which condition is evaluating false at observation time.
2. Align `canTransfer` → style → `disabled` via a single source-of-truth const so they can't diverge.
3. Consider relaxing `hasSendableWallet` the way Send/Receive were relaxed in #195: for NWC wallets, don't gate on transient `isConnected` (the underlying `pay / transfer` call auto-awaits the connect).

### LINKED ISSUE #302: bug: Transfer button stays disabled when at least one wallet can send + 2+ wallets exist
## Symptom

On the Home screen with 5 wallets configured (one of which is connected with a positive balance), the **Transfer** button is greyed out / disabled. **Send** is enabled (so the active wallet IS sendable per the per-button check).

Reproduced on: production APK `versionCode=32` (commit `f5d27bf`+ row-memo fix `17f2b6f`, build-1777849194982.apk), Pixel `37111FDJH0067B`, 2026-05-04.

Screenshot: https://raw.githubusercontent.com/BenGWeeks/lightning-piggy-mobile/main/docs/screenshots/transfer-button-bug-2026-05-04.png (TODO — attach in repo if useful).

## Root cause

`src/screens/HomeScreen.tsx:198-204`:

```ts
const hasSendableWallet = wallets.some(
  (w) =>
    (w.walletType === 'nwc' && w.isConnected) ||
    (w.walletType === 'onchain' && w.onchainImportMethod === 'mnemonic'),
);
const canTransfer = hasSendableWallet && wallets.length >= 2;
```

Two problems with this predicate:

1. **Too narrow on wallet type.** Only counts `'nwc'` (with `isConnected`) and `'onchain'` (with mnemonic import). Any other walletType (legacy LNbits direct, LNURL-pay, watch-then-paired, etc.) doesn't qualify.
2. **Diverges from the per-wallet `canSend` semantic** that `SendSheet` itself uses. `canSend` on the Home button is `hasActiveConnection && !isWatchOnly` (line 197) — this is a different rule from `hasSendableWallet`.

Result: the screen shows Send enabled (because the *active* wallet passes `canSend`), but Transfer is disabled (because the `wallets.some(...)` rule rejects all 5 wallets including the active one).

## Suggested fix

Replace the narrow `walletType` switch with a single source-of-truth helper:

```ts
function isSendableWallet(w: Wallet): boolean {
  if (w.walletType === 'nwc') return w.isConnected;
  if (w.walletType === 'onchain') return w.onchainImportMethod === 'mnemonic';
  // ... extend for any other types that can send
  return false;
}

const hasSendableWallet = wallets.some(isSendableWallet);
```

Better — derive it from each wallet's existing capability metadata. If `Wallet` doesn't already carry a `canSend` (or `walletKind === 'sendable'`) field, add one. Using a tagged type is more robust than enumerating walletType strings inline.

## Test plan

- [ ] Add a unit test in `src/screens/HomeScreen.test.tsx` (or wherever Home logic is exercised) that constructs 2+ wallets of mixed types where at least one is sendable, asserts `canTransfer === true`.
- [ ] Test on real device — Pixel with 5 wallets, NWC + on-chain + Coinos managed (mixed), confirm Transfer button enables.

## Related

- The existing logic was added to gate against single-wallet edge cases. Don't regress that — `wallets.length >= 2` should stay.
- The fix should respect parental-control sendability rules: if a wallet is locked / spending-disabled, it should NOT count as sendable, regardless of walletType.

## Severity

User-visible, blocks a primary workflow (moving funds between wallets). Medium-high.

### PR DESCRIPTION
**Pull request by BenGWeeks**
## Summary

- The Transfer button gate (`hasSendableWallet` in `HomeScreen.tsx`) required `w.isConnected` for NWC wallets, which contradicts the per-wallet `canSend` rule. The long comment by `canSend` (lines 190-195) explicitly says we don't gate Send/Receive on the transient `isConnected` flag because NWC wallets land in state with `isConnected: false` and flip true in background; gating there would dead-lock the buttons for the 2-14 s `enable()` window, or indefinitely if the WebSocket blips. The Transfer gate was doing exactly the thing we said not to do.
- Extracted the per-wallet sendability check into `isSendableWallet` (`src/utils/walletCapabilities.ts`) so the Send button and the Transfer "any wallet can send" check share one source of truth. The helper is exhaustive over `WalletType` (`'nwc' | 'onchain'`) and documents the rule for future wallet types.
- Added focused unit tests at `src/utils/walletCapabilities.test.ts` covering the per-wallet predicate plus a `canTransfer`-equivalent that locks the contract HomeScreen relies on. The headline regression case asserts that two NWC wallets with `isConnected: false` enable Transfer — exactly the user-reported scenario.

### "Tap-still-opens-sheet" claim from #199

Verified non-repro on standard React Native: `TouchableOpacity` with `disabled={true}` correctly suppresses `onPress`, and the existing `disabled={!canTransfer}` is wired correctly. The user's symptom in #199 was the Transfer button being greyed-out while Send worked — both rooted in the same `hasSendableWallet` bug. Once Transfer enables for the right wallet sets, the tap behaviour is correct by construction.

Closes #199
Closes #302

## Test plan

- [x] `npx tsc --noEmit` — passes
- [x] `npx eslint src/screens/HomeScreen.tsx src/utils/walletCapabilities.ts src/utils/walletCapabilities.test.ts` — clean (no new warnings)
- [x] `npx prettier --check` — clean
- [x] `npx jest src/utils/walletCapabilities.test.ts` — 10/10 tests pass, including the #199 / #302 regression case
- [ ] Smoke test on device: with 2+ NWC wallets configured, Transfer button is enabled immediately after app launch (does not require `isConnected` to flip true first)
- [ ] Smoke test: with 1 wallet only, Transfer remains disabled
- [ ] Smoke test: with 2+ watch-only on-chain wallets and no sendable wallet, Transfer remains disabled

## Screenshot

| Bundle running this branch on emulator-5554 (Home, no wallets) |
|---|
| emu-pr307-bundle |

The new behaviour (Transfer enabled when ≥1 wallet can send AND ≥2 wallets exist) requires at least two seeded wallets to demonstrate visually. The wallet-seeder subflow lands in #291; until then the rule is documented + unit-tested:

- `src/utils/walletCapabilities.ts → isSendableWallet()` — switch over WalletType + OnchainImportMethod with assertNever (see Copilot follow-up #307).
- `src/screens/HomeScreen.tsx:201-204` — `canTransfer = hasSendableWallet && wallets.length >= 2`.
- `src/utils/walletCapabilities.test.ts` — 20/20 unit tests covering every WalletType / OnchainImportMethod / isConnected combination.

### CODE REVIEW

File: src/utils/walletCapabilities.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ * the root cause of #199 / #302 where Transfer stayed greyed out
+ * while Send worked), on-chain wallets are sendable only when
+ * imported from a mnemonic (xpub-only wallets are watch-only and
+ * have no signing key).
+ *
+ * Keep this exhaustive when adding new `WalletType`s.
+ */
+export function isSendableWallet(w: WalletState): boolean {
+  if (w.walletType === 'nwc') return true;
+  if (w.walletType === 'onchain') return w.onchainImportMethod === 'mnemonic';
```
[the remark this answers is not in the document]
Comment (BenGWeeks) [reply src/utils/walletCapabilities.ts]: Addressed in `f6e30da` — `'generated'` is now explicitly non-sendable in a dedicated `isSendableOnchainImportMethod` switch with a comment stating *why*: no codepath actually creates a `'generated'` wallet today, so until the create/back-up/sign path lands it has no signing key. The intent is clear in code rather than implicit in the fall-through.

File: src/utils/walletCapabilities.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  return false;
+}
```
[the remark this answers is not in the document]
Comment (BenGWeeks) [reply src/utils/walletCapabilities.ts:46]: Addressed in `f6e30da` — refactored to `switch` over `WalletType` with `assertNever` (`const _exhaustive: never = ...`). Adding a new `WalletType` (or `OnchainImportMethod`) is now a compile error here rather than a silent runtime `false`.
